lib/local-links-manager/import/links.rb

```diff
@@ -1,31 +1,47 @@
 require "csv"
+require_relative "errors"
 
 module LocalLinksManager
   module Import
     class Links
+      attr_accessor :errors
+
       def initialize(local_authority)
         @local_authority = local_authority
+        @errors = []
       end
 
       def import_links(csv_string)
         updated = 0
+        index = 1 # start at 1 to account for headers
         CSV.parse(csv_string, headers: true) do |row|
-          next if row["New URL"].blank?
+          index += 1
+          new_url = row["New URL"]
+          next if new_url.blank?
 
           service = Service.find_by(lgsl_code: row["LGSL"])
           interaction = Interaction.find_by(lgil_code: row["LGIL"])
 
           next unless service && interaction
 
-          link = Link.retrieve_or_build(
+          slugs = {
             local_authority_slug: local_authority.slug,
             service_slug: service.slug,
             interaction_slug: interaction.slug,
-          )
-
-          link.update!(url: row["New URL"])
-
-          updated += 1
+          }
+          link = Link.retrieve_or_build(slugs)
+
+          begin
+            link.update!(url: new_url)
+            updated += 1
+          rescue ActiveRecord::RecordInvalid => e
+            GovukError.notify(
+              LocalLinksManager::Import::UrlValidationException.new(e.message),
+              extra: slugs.merge(link_id: link.id),
+            )
```

Comment:
Sorry for a drive-by comment.

We shouldn't use Sentry for logging validation errors in a users input as per [rfc-87](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-087-dealing-with-errors.md#2-notifications-should-be-actionable): 

```
### 2. Notifications should be actionable

Sentry notifications should be something that requires a developer of the app to do something about it. Not just a piece of information.
```

As these create error logs that aren't actionable or resolvable and may appear at any cadence.

The approach we've used to record information where we want a record of user errors is to write to the Rails logger so that the issue is available in Kibana.

Reply:
Thanks for the comment, Kevin. Sentry logging was a request on the Trello card, but in hindsight it does feel a bit odd to be logging an error the developer shouldn't do anything about.

I'll talk to Platform Health about a follow-up piece of work to swap this out for Rails logging.